### ISSUE TITLE
Custom event for each dependency

### ISSUE THREAD

**Issue by Kavenon** [dstreet/dependsOn#28]
It would be nice to have customizable trigger events for each dependency.

ex. iChecks.js uses event "ifChanged" instead of "changed" event

**Comment by dstreet** [comment]
I prefer the use of standard events here. However, it would be really easy to trigger your own custom events with the `onEnable` and `onDisable` callbacks.

**Comment by Decadence** [comment]
@dstreet Is there any way to get dependent object in this functions? `this` equals triggered dependency.

**Comment by dstreet** [comment]
@Decadence what is the use case for getting the dependency object (rather than just the last triggering dependency element) in the callback?

**Comment by Decadence** [comment]
@dstreet For example I have div as dependency subject that has many inputs which I need to disable or anything else. disable: true doesn't work on div directly so I need `$(div_object).find("input, select, textarea").prop("disabled", true)` in callback. But how I get div_object without static selector for each dependency set?

**Comment by dstreet** [comment]
Well, for that scenario, you could simply use a selector that includes many elements for the subject: `$('#div-obj > input, #div-obj > textarea').dependsOn(...)`

**Comment by Decadence** [comment]
But what if I need to do some complex logic or actions with elements, not just hide / disable elements?
Right now I handle it by this code
```js
function createRule(rule) {
    var subject = $(rule.subject);

    if (subject.length === 0) {
        return;
    }

    var rule_object = {};

    rule_object[rule.dependency] = {
        values: rule.values,
    };

    subject.dependsOn(rule_object, {
        onEnable: function () {
            disableElements(rule.subject, false);
        },
        onDisable: function () {
            disableElements(rule.subject, true);
        }
    });
}
```
But I was looking for cleanest solution.

**Comment by dstreet** [comment]
Well I could certainly pass the subject through the `onEnable` and `onDisable` callbacks.

**Comment by Decadence** [comment]
It would be nice I think :)

**Comment by dstreet** [comment]
I'm going to leave this issue closed, but I created issue #32 to handle this.

**Comment by Decadence** [comment]
Ok, thanks!
